pennylane/ops/qubit.py

```diff
@@ -1601,6 +1603,60 @@ def _matrix(cls, *params):
         return U
 
 
+class ControlledQubitUnitary(QubitUnitary):
+    r"""ControlledQubitUnitary(U, control_wires, wires)
+    Apply an arbitrary fixed unitary to ``wires`` with control from the ``control_wires``.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 1
+    * Gradient recipe: None
+
+    Args:
+        U (array[complex]): square unitary matrix
+        control_wires (Sequence[int] or int): the control wire(s)
+        wires (Sequence[int] or int): the wire(s) the unitary acts on
+
+    **Example**
+
+    The following shows how a single-qubit unitary can be applied to wire ``2`` with control on
+    both wires ``0`` and ``1``:
+
+    >>> U = np.array([[ 0.94877869,  0.31594146], [-0.31594146,  0.94877869]])
+    >>> qml.ControlledQubitUnitary(U, control_wires=[0, 1], wires=2)
+    """
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+    grad_method = None
+
+    def __init__(self, *params, wires=None, do_queue=True, control_wires=None):
+        if control_wires is None:
+            raise ValueError("Must specify control wires")
+
+        wires = Wires(wires)
+        control_wires = Wires(control_wires)
+
+        if Wires.shared_wires([wires, control_wires]):
+            raise ValueError(
+                "The control wires must be different from the wires specified to apply the unitary on"
+            )
+
+        U = params[0]
+        if len(U) != 2 ** len(wires):
+            raise ValueError(f"Input unitary must be of dimension {2 ** len(wires)}")
```

Comment:
(Minor) Just a thought: do we want to be even more explicit here?
```suggestion
            raise ValueError(f"Input unitary must have dimensions {2 ** len(wires)}x{2 ** len(wires)}.")
```

Reply:
I'll add it as a shape :+1: